src/bigint.cpp

```diff
@@ -0,0 +1,65 @@
+/**
+ * Boost Software License - Version 1.0 - August 17th, 2003
+ * Permission is hereby granted, free of charge, to any person
+ * or organization obtaining a copy of the software and 
+ * accompanying documentation covered by this license 
+ * (the "Software") to use, reproduce, display, distribute, 
+ * execute, and transmit the Software, and to prepare derivative 
+ * works of the Software, and to permit third-parties to whom the 
+ * Software is furnished to do so, all subject to the following:
+ * 
+ * The copyright notices in the Software and this entire statement, 
+ * including the above license grant, this restriction and the following 
+ * disclaimer, must be included in all copies of the Software, in whole or 
+ * in part, and all derivative works of the Software, unless such copies 
+ * or derivative works are solely in the form of machine-executable 
+ * object code generated by a source language processor.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, 
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF 
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, TITLE AND 
+ * NON-INFRINGEMENT. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR ANYONE
+ * DISTRIBUTING THE SOFTWARE BE LIABLE FOR ANY DAMAGES OR OTHER LIABILITY, 
+ * WHETHER IN CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN 
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * 
+ */
+
+#include "bigint.hpp"
+
+namespace libbig {
+	largeInt::largeInt() {
+		this->sign = 1;
+		this->number = "0";
+	}
+
+	largeInt::largeInt(int nValue) {
+		if (nValue < 0) {
+			this->sign = 0;
+			nValue *= -1;
+		} else {
+			this->sign = 1;
+		}
+		this->number = std::to_string(nValue);
+	}
+
+	largeInt::largeInt(long long nValue) {
+		if (nValue < 0) {
+			this->sign = 0;
+			nValue *= -1;
+		} else {
+			this->sign = 1;
+		}
+		this->number = std::to_string(nValue);
+	}
+
+	largeInt::largeInt(std::string nValue) {
+		if (nValue.front() == '-') {
+			this->sign = 0;
+			nValue.erase(nValue.begin());
```

Comment:
this seems to be an extra operation, as the first index becomes redundant after getting the sign of number we can just direclty copy from `(nValue.begin() + 1, nValue.end());`, there is no need to call erase on `nValue`.

Reply:
this also needs to be fixed @Antimeuh2433